store/tikv/txn_committer.go

```diff
@@ -231,21 +229,34 @@ func (c *txnCommitter) commitSingleRegion(keys [][]byte) error {
 }
 
 func (c *txnCommitter) cleanupSingleRegion(keys [][]byte) error {
-	// TODO: add batch RPC call.
-	for _, k := range keys {
-		req := &pb.Request{
-			Type: pb.MessageType_CmdCleanup.Enum(),
-			CmdCleanupReq: &pb.CmdCleanupRequest{
-				Key:          k,
-				StartVersion: proto.Uint64(c.startTS),
-			},
-		}
-		region, err := c.store.getRegion(k)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		c.store.SendKVReq(req, region)
-		// TODO: check response and retry
+	req := &pb.Request{
+		Type: pb.MessageType_CmdBatchRollback.Enum(),
+		CmdBatchRollbackReq: &pb.CmdBatchRollbackRequest{
+			Keys:         keys,
+			StartVersion: proto.Uint64(c.startTS),
+		},
+	}
+
+	region, err := c.store.getRegion(keys[0])
```

Comment:
@morefreeze The algorithm guarantees it won't be an empty slice. By the way, I will refactor it soon and use an ID to get region.